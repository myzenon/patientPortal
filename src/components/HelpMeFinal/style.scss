@import '~themes/colors.scss';
@import '~themes/mixins.scss';

@import '~themes/css/styles.scss';



.wrapper {
    position: fixed;
    left: 50%;
    top: 50%;
    z-index: 50;
    height: 580px;
    width: 837px;
    border-radius: 6px;	
    background-color: #fff;
    transform: translateX(-50%) translateY(-50%);
    border-radius: 6px;
    box-shadow: 0px 0px 10px $confirm-box-shadow-color;
    text-align: center;
}

 

.header {  
    color: #344559;
    font-family: "Nunito Sans";
    font-size: 32px;
    font-weight: 800;	
    line-height: 41px;	
    text-align: center; 
    padding-top: 50px;  
    padding-bottom: 50px;
    text-align: center; 
}
 

 
.headerTextCloseSucc{    
    text-align: right;
    display: inline; 
    cursor: pointer;
} 
  
.line {	  
    align-items: left;
    box-sizing: border-box;
    height: 2px;	
    width: 100%;	
    border: 1px solid #DBDBDB;
}

.empty30{
    width: 100%;
    height: 30px;
}
.empty60{
    width: 100%;
    height: 60px;
}

.contactNumber{ 
    font-weight: bold;
    color: #0DA4C5;
    font-family: "Nunito Sans";	
    font-size: 24px;	 
    line-height: 32px;
}


.rectangle { 
    height: 96px;	
    width: 598px;	
    border-radius: 53.5px;
    background-color: #fff;
    box-shadow: 0 2px 7px 0 rgba(0,0,0,0.38); 
    padding-right: 50px;  
    padding-top:25px;   
    display: flex;  
    color: #344559;
    font-family: "Nunito Sans";
    font-size: 24px;	
    font-weight: 800;	
    line-height: 32px;
    text-align: center; 
}
.rectangle:hover path {
    fill: #FFFFFF;
    transition: all ease 0.3s;
  }
  
.rectangle path {
    transition: all ease 0.3s;
  }
.rectangle:hover {	  
    color: #FFFFFF;
    font-family: "Nunito Sans";
    font-size: 24px;	
    font-weight: 800;	
    line-height: 32px;
    background-color: #0DA4C5; 
    text-align: center;   
}

.helpText{
    height: 32px;
    width: 564px;
    color: #344559 !important;	
    font-family: Nunito Sans;
    font-size: 24px;	
    font-weight: bold; 
    line-height: 32px; 
    text-decoration: none;
  }
 
.phoneIcon{
    height: 53px;
    width: 53px;
    margin-left: 26px;
}

.mailIcon{
    height: 38px;
    width: 51px;
    margin-left: 26px;
}

.iconAlign{
    text-align: left;  
    padding-left: 20px;
}

.iconTextlign{  
    text-align: center; 
    height: 34px;
    width: 518px; 
    font-family: "Nunito Sans";
    font-size: 24px;	
    font-weight: 800;	
    line-height: 42px;
    text-align: center; 
}

 .emailUsText{
    height: 38px;	
    width: 170px;
    color: #344559;
    font-family: "Nunito Sans";	
    font-size: 28px;	
    font-weight: 600;
    line-height: 38px;
    text-align: center; 
 }

 .subHeading{ 
    color: #344559;
    font-family: "Nunito Sans";	
    font-size: 24px;
    font-weight: 600;
    line-height: 41px;	
    text-align: center; 
 }