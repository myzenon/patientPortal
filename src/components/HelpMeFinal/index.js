import React, { Component } from 'react'
import { Button } from 'reactstrap'
import style from './style.scss'

import { Link } from 'react-router-dom'
import Mail from 'assets/mail.svg'
import Call from 'assets/call_black.svg' 
import ReactGA from 'react-ga';
ReactGA.initialize('UA-121783543-1');

export default class HelpMeFinal extends Component {
    static defaultProps = {
        onDone: () => {}
    }
    componentDidMount() {
        const page = window.location.pathname + window.location.search;
        ReactGA.pageview(page);
    }

    render() {
        return (
            <div className={style.wrapper}>
 
                 <div className={style.header}> 

                   <span
                              className={style.headerText}>
                      &nbsp;&nbsp;&nbsp;Give us a call and we'll sort it out
                      </span>
                            <span
                              className={style.headerTextCloseSucc}
                                onClick={this.props.onDone} >
                               &nbsp;
                                X
                            </span>
                            <div className={style.subHeading}>
                            (Don't worry, it's free!)
                            </div>

                    </div>

                        <div className={style.line} /> 
                        <div className={style.empty60} /> 
  
                      <div > 
                       
                       <Link  className={style.helpText} to="" >
                       <div className={style.rectangle}>
                       <span  className={style.iconAlign}> 

                       <Call className={style.phoneIcon}/>
                       </span>
                       
                       <span  className={style.iconTextlign}>

                        {process.env.ARINE_PHONE}
                        </span>
                       </div> 
                       </Link>
                
                   </div>
                   <div className={style.empty30} /> 
  
                <div className={style.emailUsText}> or email us at </div>
                <div className={style.empty30} /> 
  
                       <div > 
                       
                       <a  className={style.helpText} href={"mailto:" + process.env.ARINE_EMAIL + '"'} target="_blank">
                       <div className={style.rectangle}>
                       <span  className={style.iconAlign}> <Mail className={style.mailIcon}/>
                       </span>
                       
                       <span  className={style.iconTextlign}>
                       {process.env.ARINE_EMAIL}
                       </span>
                       </div> 
                       </a>
                
                   </div>
 

               
            </div>
        )
    }
}
