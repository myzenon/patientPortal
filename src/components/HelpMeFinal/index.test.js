import React from 'react'
import HelpMeFinal from './index'
import ReactGA from 'react-ga';

   // This should be part of your setup
   ReactGA.initialize('UA-121783543-1', {
       'debug': true
     });;

jest.mock("react-ga")
test('renders correctly', () => {
    const wrapper = render (
        <HelpMeFinal />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
})

test('clicks done is call props ondone fn', done => {
    const onDone = () => {
        done()
    }
    const wrapper = mount (
        <HelpMeFinal onDone={onDone} />
    )
    wrapper.find('button').simulate('click')
})