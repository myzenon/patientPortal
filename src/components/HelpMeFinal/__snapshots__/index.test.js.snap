// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  class="wrapper"
>
  <div
    class="logoWrapper"
  >
    <img
      alt="Arine"
      class="logo"
      src="logo.png"
    />
  </div>
  <br />
  <p>
    Give us a call and we'll sort it out.
  </p>
  <p>
    (Don't worry, it's free!)
  </p>
  <p
    class="phoneNumber"
  />
  <p>
     or email us at 
  </p>
  <p
    class="phoneNumber"
  >
       
  </p>
  <p>
    <button
      class="button btn btn-secondary"
      id="linkDoneMeFindHPID"
      type="button"
    >
      Done
    </button>
  </p>
</div>
`;
